core/variant/variant_parser.cpp

```diff
@@ -1401,12 +1425,21 @@ Error VariantParser::parse(Stream *p_stream, Variant &r_ret, String &r_err_str,
 
 //////////////////////////////////////////////////////////////////////////////////
 //////////////////////////////////////////////////////////////////////////////////
-////////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////
 
 static String rtosfix(double p_value) {
 	if (p_value == 0.0) {
 		return "0"; //avoid negative zero (-0) being written, which may annoy git, svn, etc. for changes when they don't exist.
 	} else {
+		if (isnan(p_value)) {
+			return "nan";
+		} else if (isinf(p_value)) {
+			if (p_value > 0) {
+				return "inf";
+			} else {
+				return "inf_neg";
```

Comment:
Is there a specific reason why this is `inf_neg` and not `-inf`?

Reply:
Looks like it's to avoid having this be parsed as a `TK_NUMBER` token instead of `TK_IDENTIFIER`.